docs/markdown/Fs-module.md

```diff
@@ -194,3 +194,32 @@ Returns the last component of the path, dropping the last part of the suffix
 fs.stem('foo/bar/baz.dll')  # baz
 fs.stem('foo/bar/baz.dll.a')  # baz.dll
 ```
+
+### relative_to
+
+*since 0.57.0*
+
+Given two absolute paths, returns a version of the first path relative to the second.
+If a path is given as the 'within' argument, the function will return the first path
+unchanged if it is not inside 'within'.
```

Comment:
Why does within need to be a path? It's contextually meaningless unless it is a Windows drive letter, and in that case it can just as well be a kwarg to return the first path unchanged "if it cannot be rendered as a relative path".

Reply:
Since @bonzini was proposing the `within` argument here 

- https://github.com/mesonbuild/meson/issues/2184#issuecomment-674451979

he probably can see useful use cases to justify this

Reply:
In this other comment https://github.com/mesonbuild/meson/pull/7908#discussion_r577851659,  @bonzini suggests to have both `within` and `allow_absolute`. `within` is convenient for his examples in that comment.

Reply:
`if_within` needs to be a path because you may want a relative directory as long as it is within a specific subdirectory, and otherwise error or return the original absolute path. For further functional discussion I would suggest this conversation https://github.com/mesonbuild/meson/pull/7908#discussion_r555361018